Preprocess .asm files

emusand opened the pull request:
Assembler files with file extension ".asm" are now preprocessed.

The following assembly file extensions were hardcoded in Bazel:
 * ".s" and ".asm": Assumed to be generated by a compiler, and not
   preprocessed.
 * ".S": Assumed to be handwritten, and preprocessed.

There are Bazel users with large codebases where all handwritten
assembly files use file extension ".asm". ".asm" files are now treated
in the same way as ".S".

See https://github.com/bazelbuild/bazel/issues/4856

mhlopko commented:
This looks good to me, let me investigate if this will need to follow incompatible changes policy, and also to see how much will this break Google :)

hlopko commented:
It breaks a bit of Google, but I'm quite successful with migrating. I think this will not make it for 0.21, but 0.22 will very likely have this pr in.

emusand commented:
Ok, thanks for the help! 0.22 is fine for us. We will continue using our own bazel for until then.

emusand commented:
What is the status of this commit? Is there any chance that it will be included in 0.23?

emusand commented:
Any updates on this commit? What is the issue that blocks this from being merged? Is there anything I can do to help?

hlopko commented:
We will very likely accept it (and migrate Google), but @oquenchil is on vacation until next week. This change will have to follow https://docs.bazel.build/versions/master/backward-compatibility.html (good resource is also https://www.bazel.build/breaking-changes-guide.html).

emusand commented:
Thanks for the information!

I am trying to add a --incompatible_preprocess_asm_files option. The problem is the option needs to change values of public static final constants in CppFileTypes. Replacing these constants with getters that are aware of the option value seems to be a big change.

Would you accept this solution, or is it too dirty?
- Keep the constants in CppFileTypes, but let them read system properties instead.
- Parse the option in the client, and convert it to system properties.

philwo commented:
@emusand I think something went wrong with the last push - it contains a lot of commits that are unrelated to your change. Can you `git pull --rebase origin master` your branch and force push your PR?

emusand commented:
I am sorry for my previous push. My branch was not rebased properly, but I have fixed it now.

I have tried to follow the incompatible changes process. I have added a --[no]incompatible_preprocess_asm_files option, and written a GitHub issue on it: #8055. (But I could not add label incompatible-change. Can you help me with this?)

The --[no]incompatible_preprocess_asm_files is a bit special. The option needs to change values of constants in class CppFileTypes. I added the option as a start-up option to the client. The client then converts it to a Java system property. Class CppFileTypes reads this property, and sets the constants accordingly. Can you review the commit and check if you think that this solution is OK? Then I will have a look at the failing tests in CI, and polish the commit.

ulfjack commented:
It would be better for it not to be a startup option. Is it possible to have two constants in CppFileTypes and decide which to use in the code that references them?

emusand commented:
Thanks for the comment UIf!

I will try to remove the start-up option and find another way to determine
which assembly file extensions that will be preprocessed.

Den ons 2 okt. 2019 kl 12:46 skrev Ulf Adams <notifications@github.com>:

> It would be better for it not to be a startup option. Is it possible to
> have two constants in CppFileTypes and decide which to use in the code that
> references them?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bazelbuild/bazel/pull/6919?email_source=notifications&email_token=AB43UKJDDETIAMYXDA27QW3QMR3Y7A5CNFSM4GKH3VGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAEKREQ#issuecomment-537438354>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB43UKKKDRU6S4SQPMZHUALQMR3Y7ANCNFSM4GKH3VGA>
> .
>

aiuto commented:
Ping to author and reviewers.

meisterT commented:
Closing for lack of activity. Please ping if you want to reopen.
